Packages/vcs/Lib/vtk_ui/text.py

```diff
@@ -87,45 +186,146 @@ def __init__(self, interactor, string, movable=False, on_move=None, on_drag=None
         else:
             self.actor = text_actor(string, fgcolor, size, font)
 
-        widget.SetTextActor(self.actor)
+        widget = WidgetReprShim(interactor, self.actor)
 
         super(Label, self).__init__(interactor, widget)
 
-        #self.widget.ResizableOff()
         self.movable = movable
         self.action = on_click
-        self.release_action = on_release
         self.move_action = on_move
         self.dragged = on_drag
 
-        self.left, self.top = left, top
-        self.top_offset = 0
-
-        # Assigned by Widget.__init__
-        self.repr.MovingOff()
-        self.repr.PickableOff()
-        self.repr.SetShowBorderToOff()
-
         self.actor.SetTextScaleModeToNone()
+        self.actor.SetUseBorderAlign(False)
+        self.actor.VisibilityOff()
 
-        # Map events to draggable actions, because standard events aren't propagated
-        self.add_event_handler("StartInteractionEvent", self.drag_clicked)
-        self.add_event_handler("InteractionEvent", self.drag_moved)
-        self.add_event_handler("EndInteractionEvent", self.drag_released)
-        self.add_event_handler("StartInteractionEvent", self.click)
-        self.add_event_handler("EndInteractionEvent", self.release)
+        self.left = left
+        self.top = top
 
         self.register()
 
-    def get_text(self):
-        return self.repr.GetText()
+    def text():
+        doc = "Property that provides access to the text actor's input"
+
+        def fget(self):
+            return self.get_text()
+
+        def fset(self, value):
+            self.set_text(value)
+
+        return locals()
+    text = property(**text())
+
+    def x():
+        doc = "The x coordinate of the text actor."
+
+        def fget(self):
+            return self.actor.GetPosition()[0]
+
+        def fset(self, value):
+            self.actor.SetPosition(value, self.y)
+        return locals()
+    x = property(**x())
+
+    def y():
+        doc = "The y coordinate of the text actor."
+
+        def fget(self):
+            return self.actor.GetPosition()[1]
+
+        def fset(self, value):
+            self.actor.SetPosition(self.x, value)
+        return locals()
+    y = property(**y())
+
+    def left():
```

Comment:
`left` and `top` are more than just convenient; they're cleaning up a mess I made when I first wrote most of the UI code. I was new to VTK, and I like coordinate systems that are in the fourth quadrant better (blame my high school Visual Basic class). I used left/top for everything, and flipped the Y axis so I could measure things in distance from the top of the window instead of distance from the bottom. This has some pluses and minuses (really handy for placing toolbars and buttons on the screen, for example; also, I wanted to specify things in absolute numbers, because the UI should stick to sides instead of drifting towards the middle when you resize the window), but it got really annoying to be constantly adjusting for justifications in other chunks of code, so I decided to just use this as the single source of truth and calculate the left/top based off of the current justification and the x/y position. This was super useful.